docs/upgrading/android-csproj.md

```diff
@@ -0,0 +1,350 @@
+
+```xml
+<PropertyGroup>
+  <!-- Used in legacy Xamarin.Android projects -->
+  <AndroidSupportedAbis>armeabi-v7a;arm64-v8a;x86;x86_64</AndroidSupportedAbis>
+</PropertyGroup>
+```
+
+Instead use .NET's concept of [runtime identifiers][rids]:
+
+```xml
+<PropertyGroup>
+  <!-- Used going forward in .NET -->
+  <RuntimeIdentifiers>android-arm;android-arm64;android-x86;android-x64</RuntimeIdentifiers>
+</PropertyGroup>
+```
+
+`$(AndroidUseIntermediateDesignerFile)` will be `True` by default.
+
+`$(AndroidBoundExceptionType)` will be `System` by default.  This will
+[alter the types of exceptions thrown from various methods][abet-sys] to
+better align with existing .NET 6 semantics, at the cost of compatibility with
+previous Xamarin.Android releases.
+
+`$(AndroidClassParser)` will be `class-parse` by default. `jar2xml`
+will not be supported.
+
+`$(AndroidDexTool)` will be `d8` by default. `dx` will not be
+supported.
+
+`$(AndroidCodegenTarget)` will be `XAJavaInterop1` by default.
+`XamarinAndroid` will not be supported.
+
+`$(AndroidManifest)` will default to `AndroidManifest.xml` in the root
+of projects as `Properties\AssemblyInfo.cs` is no longer used in
+short-form MSBuild projects. `Properties\AndroidManifest.xml` will
+also be detected and used if it exists to ease migration.
+
+`$(DebugType)` will be `portable` by default. `full` and `pdbonly`
+will not be supported.
+
+`$(MonoSymbolArchive)` will be `False`, since `mono-symbolicate` is
+not yet supported.
+
+If Java binding is enabled with `@(InputJar)`, `@(EmbeddedJar)`,
+`@(LibraryProjectZip)`, etc. then `$(AllowUnsafeBlocks)` will default
+to `True`.
+
+Referencing an Android Wear project from an Android application will
+not be supported:
+
+```xml
+<ProjectReference Include="..\Foo.Wear\Foo.Wear.csproj">
+  <IsAppExtension>True</IsAppExtension>
+</ProjectReference>
+```
+
+[rids]: https://docs.microsoft.com/dotnet/core/rid-catalog
+[abet-sys]: https://github.com/xamarin/xamarin-android/issues/4127
+
+## Default file inclusion
+
+Default Android related file globbing behavior is defined in [`AutoImport.props`][autoimport].
+This behavior can be disabled for Android items by setting `$(EnableDefaultAndroidItems)` to `false`, or
+all default item inclusion behavior can be disabled by setting `$(EnableDefaultItems)` to `false`.
+
+[autoimport]: https://github.com/dotnet/designs/blob/4703666296f5e59964961464c25807c727282cae/accepted/2020/workloads/workload-resolvers.md#workload-props-files
+
+## Runtime behavior
+
+There is some behavioral changes to the `String.IndexOf()` method in
+.NET 5 and higher on different platforms, see details [here][indexof].
+
+[indexof]: https://docs.microsoft.com/dotnet/standard/globalization-localization/globalization-icu
+
+## Linker (ILLink)
+
+.NET 5 and higher has new [settings for the linker][linker]:
+
+* `<PublishTrimmed>true</PublishTrimmed>`
+* `<TrimMode>copyused</TrimMode>` - Enable assembly-level trimming.
```

Comment:
We found recently `CopyUsed` is broken: https://github.com/xamarin/xamarin-android/issues/6625

We didn't have a clear solution -- but it is a bad settting in general for mobile, it would make apps much larger?

Maybe we just don't mention it for now?